test/e2e/far_e2e_test.go

```diff
@@ -48,55 +47,66 @@ const (
 	skipOOSREnvVarName = "SKIP_OOST_REMEDIATION_VERIFICATION"
 )
 
-var remediationTimes []time.Duration
+var (
+	stopTesting      bool
+	remediationTimes []time.Duration
+)
 
 var _ = Describe("FAR E2e", func() {
 	var (
 		fenceAgent, nodeIdentifierPrefix string
 		testShareParam                   map[v1alpha1.ParameterName]string
 		testNodeParam                    map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string
-		selectedNode                     *corev1.Node
-		nodeName                         string
-		pod                              *corev1.Pod
-		startTime, nodeBootTimeBefore    time.Time
 		err                              error
 	)
-	BeforeEach(func() {
-		// create FAR CR spec based on OCP platformn
-		clusterPlatform, err := e2eUtils.GetClusterInfo(configClient)
-		Expect(err).ToNot(HaveOccurred(), "can't identify the cluster platform")
-		log.Info("Begin e2e test", "Cluster name", string(clusterPlatform.Name), "PlatformType", string(clusterPlatform.Status.PlatformStatus.Type))
-
-		switch clusterPlatform.Status.PlatformStatus.Type {
-		case configv1.AWSPlatformType:
-			fenceAgent = fenceAgentAWS
-			nodeIdentifierPrefix = nodeIdentifierPrefixAWS
-			By("running fence_aws")
-		case configv1.BareMetalPlatformType:
-			fenceAgent = fenceAgentIPMI
-			nodeIdentifierPrefix = nodeIdentifierPrefixIPMI
-			By("running fence_ipmilan")
-		default:
-			Skip("FAR haven't been tested on this kind of cluster (non AWS or BareMetal)")
-		}
+	When("trying to identify cluster platform", func() {
```

Comment:
Why did you switch from BeforeEach to When/It/stopTesting?

Reply:
I thought that it was redundant to fetch and print the cluster information (e.g., Platform) and agent parameter values per every e2e test (4 times). Testing and fetching this information in the beginning with independent tests saves (minimal to zero) time, and removes compatibility issues with the agent per every test.